crypto-ffi/bindings/jvm/src/test/kotlin/com/wire/crypto/MLSTest.kt

```diff
@@ -293,6 +299,48 @@ class MLSTest {
         }
     }
 
+    @Test
+    fun givenTransactionRunsSuccessfully_thenShouldBeAbleToFinishOtherTransactions() = runTest {
+        val coreCrypto = initCc()
+        val someWork = Job()
+        val firstTransactionJob = launch {
+            coreCrypto.transaction {
+                someWork.complete()
+            }
+        }
+        firstTransactionJob.join()
+
+        var didRun = false
+        val secondTransactionJob = launch {
+            coreCrypto.transaction {
+                didRun = true
+            }
+        }
+        secondTransactionJob.join()
+        assertTrue { didRun }
+    }
+
+    @Test
+    fun givenTransactionIsCancelled_thenShouldBeAbleToFinishOtherTransactions() = runTest {
+        val coreCrypto = initCc()
+
+        val firstTransactionJob = launch {
+            coreCrypto.transaction {
+                this@launch.cancel()
+            }
+        }
+        firstTransactionJob.join()
+
+        var didRun = false
+        val secondTransactionJob = launch {
+            coreCrypto.transaction {
+                didRun = true
+            }
+        }
+        secondTransactionJob.join()
+        assertTrue { didRun }
+    }
```

Comment:
Before the fix, `secondTransactionJob` wouldn't complete and the test would timeout.